### DISCUSSION TITLE
Interfacing DATAVERSE

### DISCUSSION THREAD

**Discussion by umbz** [bitfoundation/bitplatform#7221] [Ideas]
Hi all, I would like to use this platform to build a portal starting from the "Admin template" which seems to suit me. 
However, I would like to use the power platform dataverse as database. **ASp.core** does not seem to me to have native drivers for such platform while there are various "clients" that can be used such as this one: https://github.com/microsoft/PowerPlatform-DataverseServiceClient

In general, the database is only reachable through api (ODATA v4).

What do you think of this idea?

However, I do not know what tables bitplatform uses to manage its data structures, where can I find the information?

**Comment by yasmoradi** [comment]
Greetings
AdminPanel needs database tables for the followings:
1- aspnetcore identity tables such as users and roles
2- data protection api keys table to scale out aspnetcore data protection api module.

You can easily remove codes related to 2nd item.

For the first one, you've to provide your authentication mechanism in IdentityController, then you can remove aspnetcore identity nuget packages and codes (I assume that you need authentication)

The other option is keep using aspnetcore identity and using data verse for the rest of the application.

Note that products, categories pages are using database tables for demonstration purposes only and you can fetch data from universally anywhere!
